dandi/metadata.py

```diff
@@ -95,9 +95,100 @@ def get_metadata(path):
     return meta
 
 
+def _parse_iso8601(age):
```

Comment:
here are the regex patterns we use in reproschema. for at least iso8601, we should cover the formal patterns for duration definition.

https://github.com/ReproNim/reproschema/blob/master/validation/reproschema-shacl.ttl#L327

Reply:
i'm checking the regex as it seems to fail for something specific, but in general the following age patterns should be supported:

https://en.wikipedia.org/wiki/ISO_8601#Durations

```
<duration>  # This would be pure age
<start>/<duration>  # This is for encoding date of birth/gestation
```

Reply:
sorry the error was weird, but this should help for the duration pattern:

```python
In [20]: pat = "^P(?!$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d
    ...: +)?D)?(T(?=\\d)(\\d+(?:\\.\\d+)?H)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?S)?)?$"

In [21]: re.match(pat, "P149D")
Out[21]: <re.Match object; span=(0, 5), match='P149D'>

In [22]: re.findall(pat, "P149D")
Out[22]: [('', '', '', '149D', '', '', '', '')]

In [26]: re.findall(pat, "P0Y2M1W149DT23H3M6S")
Out[26]: [('0Y', '2M', '1W', '149D', 'T23H3M6S', '23H', '3M', '6S')]
```

Reply:
ok, can move to this, thanks!

Reply:
it doesn't cover "P149.D" - is that ok?

Reply:
it also doesn't change "P149.0D" to "P149D", but that's perhaps is better

Reply:
it's not valid iso8601 so that should be ok. i think anything that is close but not ok, could be converted and then verified with the regex. i think my above comment was mostly for ensuring that the age format we use corresponds to one of those two iso8601 cases.

1. if input is valid for either case:
   add to age as is
2. try converting and repeat step 1.
3. if all conversions fail, raise an error and ask user to fix.

i also suspect that we need to somehow indicate in nwb the age reference (right now we are using gestat* to determine alternate or assigning birth reference). the ideal will be when we don't need to modify at all, just validate iso8601 + reference.

Reply:
I would disagree that the pattern you posted covers all valid ISO8601 duration values and only valid ones. At least not how I understand the wikipedia description, e.g. "P0,5Y" is a valid value that is not covered, and "P0.6Y2D" is not a valid value that would work with the pattern.

Reply:
It's just not clear to me should I be fixing values like this "P0.6Y2D" or we just keep them